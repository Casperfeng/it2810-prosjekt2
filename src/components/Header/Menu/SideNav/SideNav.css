.sidenav {
  z-index: 99;
  display: flex;
  flex-direction: column;
  background-color: #222222;
  position: absolute;
  right: 0;
  overflow: auto;
  width: 300px;
  height: 100vh;
}

.category {
  color: white;
  font-size: 28px;
  padding-top: 50px;
  align-self: start;
  margin-left: 55px;
  font-weight: bold;
}

.option {
  font-size: 18px;
  margin: 10px auto 10px 70px;
  color: white;
}

.option:hover {
  cursor: pointer;
  color: #ffd520;
  font-weight: bold;
}

.option-active {
  color: #ffd520;;
}

.option-active::before {
  content: "\2022";  /* Add bullet point */
  font-weight: bold;
  display: inline-block; /* Needed to add space between the bullet and the text */
  width: 1em; 
  margin-left: -1em; 
}

.option-active:hover {
  cursor: default;
  font-weight: normal;
}

.empty-box {
  padding: 25px;
}