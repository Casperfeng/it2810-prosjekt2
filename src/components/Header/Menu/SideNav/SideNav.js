import React from 'react'
import './SideNav.css';

export default function SideNav(props) {
  return (
    <div className='sidenav'>

      <div className='menu-item'>Bilder</div>
      <p className='menu-option' onClick={() => props.onSvgIndexChanged(0)}>Høst</p>
      <p className='menu-option' onClick={() => props.onSvgIndexChanged(1)} > Sommer</p>
      <p className='menu-option' onClick={() => props.onSvgIndexChanged(2)} > Vinter</p>
      <div className='menu-item'>Lyd</div>
      <p className='menu-option' onClick={() => props.onAudioIndexChanged(0)} >Japansk</p>
      <p className='menu-option' onClick={() => props.onAudioIndexChanged(1)}>Latter</p>
      <p className='menu-option' onClick={() => props.onAudioIndexChanged(2)}>Musikk</p>
      <div className='menu-item'>Tekst</div>
      <p className='menu-option' onClick={() => props.onPoemIndexChanged(0)} >Katt</p>
      <p className='menu-option' onClick={() => props.onPoemIndexChanged(1)}>Hund</p>
      <p className='menu-option' onClick={() => props.onPoemIndexChanged(2)}>Kjærlighet</p>
    </div>
  )
}
