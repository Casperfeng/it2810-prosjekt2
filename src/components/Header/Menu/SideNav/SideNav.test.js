import React from 'react';
import SideNav from './SideNav';
import renderer from 'react-test-renderer';

it('renders SideNav correctly', () => {
  const tree = renderer.create(<SideNav />).toJSON();
  expect(tree).toMatchSnapshot();
});
