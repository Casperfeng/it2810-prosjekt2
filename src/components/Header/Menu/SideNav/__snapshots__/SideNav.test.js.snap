// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideNav correctly 1`] = `
<div
  className="sidenav"
>
  <div
    className="category"
  >
    Image
  </div>
  <p
    className="option "
    onClick={[Function]}
  >
    Autumn
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Summer
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Winter
  </p>
  <div
    className="category"
  >
    Sound
  </div>
  <p
    className="option "
    onClick={[Function]}
  >
    Japanese
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Laughter
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Music
  </p>
  <div
    className="category"
  >
    Poem
  </div>
  <p
    className="option "
    onClick={[Function]}
  >
    Cat
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Dog
  </p>
  <p
    className="option "
    onClick={[Function]}
  >
    Love
  </p>
  <div
    className="empty-box"
  />
</div>
`;
